website/oauth/models/__init__.py

```diff
@@ -443,3 +443,63 @@ def absolute_api_v2_url(self):
     # used by django and DRF
     def get_absolute_url(self):
         return self.absolute_api_v2_url
+
+class ApiOAuth2PersonalToken(StoredObject):
+    """
+    Store information about recognized OAuth2 scopes. Only scopes registered under this database model can
+        be requested by third parties.
+    """
+    _id = fields.StringField(primary=True,
+                             default=lambda: str(ObjectId()))
+
+    token_id = fields.StringField(default=lambda: uuid.uuid4().hex,  # Not *guaranteed* unique, but very unlikely
```

Comment:
`uuid` is fine for just an identifier, but the actual access token should be even harder to guess. Most access tokens generated by CAS appear to be [70+ characters](https://github.com/CenterForOpenScience/cas-overlay/blob/develop/OAuth2.ipynb) usually. (I'll let you work out the specifics)

Use `website.security.random_string` here (the same function that `generate_client_secret` is based on) and specify the desired length.